testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/common/HttpsProxy.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.tests.product.launcher.env.common;
+
+import com.google.common.collect.ImmutableList;
+import com.google.inject.Inject;
+import io.trino.tests.product.launcher.docker.DockerFiles;
+import io.trino.tests.product.launcher.docker.DockerFiles.ResourceProvider;
+import io.trino.tests.product.launcher.env.Environment;
+
+import java.nio.file.Path;
+import java.util.List;
+
+import static io.trino.tests.product.launcher.env.common.HttpProxy.PROXY;
+import static java.util.Objects.requireNonNull;
+import static org.testcontainers.utility.MountableFile.forHostPath;
+
+public class HttpsProxy
+        implements EnvironmentExtender
+{
+    private final ResourceProvider configDir;
+    private final HttpProxy httpProxy;
+
+    @Inject
+    public HttpsProxy(DockerFiles dockerFiles, HttpProxy httpProxy)
+    {
+        this.configDir = requireNonNull(dockerFiles, "dockerFiles is null")
+                .getDockerFilesHostDirectory("common/https-proxy");
+        this.httpProxy = requireNonNull(httpProxy, "httpProxy is null");
+    }
+
+    @Override
+    public void extendEnvironment(Environment.Builder builder)
+    {
+        builder.configureContainer(PROXY, proxyContainer -> proxyContainer
+                .withCopyFileToContainer(
+                        forHostPath(configDir.getPath("httpd.conf")),
```

Comment:
could you please move the source file to the same location `/usr/local/apache2/conf` in the git repository, as it will be mapped in the container?

Reply:
imho,  moving to just subdir proxy will be just enough as there are no others files that need to be transferred to other dirs within proxy container, won't be just too overeagerly?   what do others think?

Reply:
later we can consider to bind folder from git repository to folder in container, 
and the structure will follow.

Reply:
conf directory contains also other files that we would need to incorporate to git, for now I don't see any advantage, let's keep it simple